// ngx-slide-toggle/__ivy_ngcc__/esm2015/ngx-slide-toggle.js
import { Component, EventEmitter, Input, NgModule, Output, ViewChild } from '@angular/core';
import { CommonModule } from '@angular/common';

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
import * as ɵngcc0 from '@angular/core';

const _c0 = ["materialSlideToggle"];
let nextUniqueId = 0;
class MaterialToggleComponent {
    constructor() {
        this._uniqueId = `material-slide-toggle-${++nextUniqueId}`;
        this.name = null;
        this.id = this._uniqueId;
        // tslint:disable:no-input-rename
        this.ariaLabel = null;
        this.ariaLabelledby = null;
        this.change = new EventEmitter();
    }
    /**
     * @param {?} event
     * @return {?}
     */
    onChange(event) {
        event.stopPropagation();
        this.change.emit(this.inputElement.nativeElement.checked);
    }
    /**
     * @param {?} event
     * @return {?}
     */
    onClick(event) {
        event.stopPropagation();
    }
}
MaterialToggleComponent.ɵfac = function MaterialToggleComponent_Factory(t) { return new (t || MaterialToggleComponent)(); };
MaterialToggleComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: MaterialToggleComponent, selectors: [["material-slide-toggle"]], viewQuery: function MaterialToggleComponent_Query(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵviewQuery(_c0, 1);
    } if (rf & 2) {
        let _t;
        ɵngcc0.ɵɵqueryRefresh(_t = ɵngcc0.ɵɵloadQuery()) && (ctx.inputElement = _t.first);
    } }, inputs: { name: "name", id: "id", ariaLabel: ["aria-label", "ariaLabel"], ariaLabelledby: ["aria-labelledby", "ariaLabelledby"], checked: "checked", disabled: "disabled" }, outputs: { change: "change" }, decls: 4, vars: 11, consts: [[1, "material-switch"], ["type", "checkbox", 3, "id", "checked", "disabled", "click", "change"], ["materialSlideToggle", ""], [1, "label-primary"]], template: function MaterialToggleComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵelementStart(0, "div", 0);
        ɵngcc0.ɵɵelementStart(1, "input", 1, 2);
        ɵngcc0.ɵɵlistener("click", function MaterialToggleComponent_Template_input_click_1_listener($event) { return ctx.onClick($event); })("change", function MaterialToggleComponent_Template_input_change_1_listener($event) { return ctx.onChange($event); });
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelement(3, "label", 3);
        ɵngcc0.ɵɵelementEnd();
    } if (rf & 2) {
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵclassProp("disabled", ctx.disabled);
        ɵngcc0.ɵɵproperty("id", ctx.id)("checked", ctx.checked)("disabled", ctx.disabled);
        ɵngcc0.ɵɵattribute("name", ctx.name)("aria-label", ctx.ariaLabel)("aria-labelledby", ctx.ariaLabelledby);
        ɵngcc0.ɵɵadvance(2);
        ɵngcc0.ɵɵclassProp("disabled", ctx.disabled);
        ɵngcc0.ɵɵattribute("for", ctx.id);
    } }, styles: [".material-switch[_ngcontent-%COMP%] {\n      display: inline-flex;\n      margin: 1em;\n      vertical-align: middle;\n      width: auto;\n    }\n\n    .material-switch[_ngcontent-%COMP%] > input[type=\"checkbox\"][_ngcontent-%COMP%] {\n      display: none;\n    }\n\n    .material-switch[_ngcontent-%COMP%] > label[_ngcontent-%COMP%] {\n      cursor: pointer;\n      height: 0px;\n      position: relative;\n      width: 40px;\n    }\n\n    .material-switch[_ngcontent-%COMP%] > label[_ngcontent-%COMP%]::before {\n      background: #000000;\n      box-shadow: inset 0px 0px 10px rgba(0, 0, 0, 0.5);\n      border-radius: 8px;\n      content: '';\n      height: 16px;\n      margin-top: -8px;\n      position: absolute;\n      opacity: 0.3;\n      transition: all 0.4s ease-in-out;\n      width: 40px;\n    }\n\n    .material-switch[_ngcontent-%COMP%] > label[_ngcontent-%COMP%]::after {\n      background: #ffffff;\n      border-radius: 16px;\n      box-shadow: 0px 0px 5px rgba(0, 0, 0, 0.3);\n      content: '';\n      height: 24px;\n      left: -4px;\n      margin-top: -8px;\n      position: absolute;\n      top: -4px;\n      transition: all 0.3s ease-in-out;\n      width: 24px;\n    }\n\n    .material-switch[_ngcontent-%COMP%] > input[type=\"checkbox\"][_ngcontent-%COMP%]:checked + label[_ngcontent-%COMP%]::before {\n      background: #3f51b5;\n      opacity: 0.5;\n    }\n\n    .material-switch[_ngcontent-%COMP%] > input[type=\"checkbox\"][_ngcontent-%COMP%]:checked + label[_ngcontent-%COMP%]::after {\n      background: #3f51b5;\n      left: 20px;\n    }\n\n    .disabled[_ngcontent-%COMP%] {\n      cursor: default;\n    }"] });
/** @nocollapse */
MaterialToggleComponent.ctorParameters = () => [];
MaterialToggleComponent.propDecorators = {
    "name": [{ type: Input },],
    "id": [{ type: Input },],
    "checked": [{ type: Input },],
    "disabled": [{ type: Input },],
    "ariaLabel": [{ type: Input, args: ['aria-label',] },],
    "ariaLabelledby": [{ type: Input, args: ['aria-labelledby',] },],
    "change": [{ type: Output },],
    "inputElement": [{ type: ViewChild, args: ['materialSlideToggle',] },],
};
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && ɵngcc0.ɵsetClassMetadata(MaterialToggleComponent, [{
        type: Component,
        args: [{
                selector: 'material-slide-toggle',
                template: `
    <div class="material-switch">
      <input #materialSlideToggle [id]="id" [attr.name]="name" type="checkbox" [attr.aria-label]="ariaLabel" [attr.aria-labelledby]="ariaLabelledby"
        (click)="onClick($event)" (change)="onChange($event)" [checked]="checked" [disabled]="disabled" [class.disabled]="disabled"
      />
      <label [attr.for]="id" [class.disabled]="disabled" class="label-primary"></label>
    </div>
  `,
                styles: [`
    .material-switch {
      display: inline-flex;
      margin: 1em;
      vertical-align: middle;
      width: auto;
    }

    .material-switch>input[type="checkbox"] {
      display: none;
    }

    .material-switch>label {
      cursor: pointer;
      height: 0px;
      position: relative;
      width: 40px;
    }

    .material-switch>label::before {
      background: #000000;
      box-shadow: inset 0px 0px 10px rgba(0, 0, 0, 0.5);
      border-radius: 8px;
      content: '';
      height: 16px;
      margin-top: -8px;
      position: absolute;
      opacity: 0.3;
      transition: all 0.4s ease-in-out;
      width: 40px;
    }

    .material-switch>label::after {
      background: #ffffff;
      border-radius: 16px;
      box-shadow: 0px 0px 5px rgba(0, 0, 0, 0.3);
      content: '';
      height: 24px;
      left: -4px;
      margin-top: -8px;
      position: absolute;
      top: -4px;
      transition: all 0.3s ease-in-out;
      width: 24px;
    }

    .material-switch>input[type="checkbox"]:checked+label::before {
      background: #3f51b5;
      opacity: 0.5;
    }

    .material-switch>input[type="checkbox"]:checked+label::after {
      background: #3f51b5;
      left: 20px;
    }

    .disabled {
      cursor: default;
    }
  `]
            }]
    }], function () { return []; }, { name: [{
            type: Input
        }], id: [{
            type: Input
        }], ariaLabel: [{
            type: Input,
            args: ['aria-label']
        }], ariaLabelledby: [{
            type: Input,
            args: ['aria-labelledby']
        }], change: [{
            type: Output
        }], checked: [{
            type: Input
        }], disabled: [{
            type: Input
        }], inputElement: [{
            type: ViewChild,
            args: ['materialSlideToggle']
        }] }); })();

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
class SlideToggleModule {
}
SlideToggleModule.ɵfac = function SlideToggleModule_Factory(t) { return new (t || SlideToggleModule)(); };
SlideToggleModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: SlideToggleModule });
SlideToggleModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ imports: [[CommonModule]] });
/** @nocollapse */
SlideToggleModule.ctorParameters = () => [];
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(SlideToggleModule, { declarations: function () { return [MaterialToggleComponent]; }, imports: function () { return [CommonModule]; }, exports: function () { return [MaterialToggleComponent]; } }); })();
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && ɵngcc0.ɵsetClassMetadata(SlideToggleModule, [{
        type: NgModule,
        args: [{
                imports: [CommonModule],
                declarations: [MaterialToggleComponent],
                exports: [MaterialToggleComponent]
            }]
    }], null, null); })();

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
/**
 * Generated bundle index. Do not edit.
 */

export { MaterialToggleComponent, SlideToggleModule };
